android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java

```diff
@@ -0,0 +1,165 @@
+
+    GlobalParamRepository globalParamRepository;
+
+    CertificateManagerService certificateManagerService;
+
+    LocalClientCryptoServiceImpl clientCryptoFacade;
+
+    CryptoManagerService cryptoManagerServiceImpl;
+
+    ClientCryptoManagerService clientCryptoManagerService;
+
+
+
+    @Inject
+    public DocumentCategoryApi(RegistrationService registrationService,FileSignatureDao fileSignatureRepository,GlobalParamRepository globalParamRepository,MasterDataService masterDataService,CertificateManagerService certificateManagerService,CryptoManagerService cryptoManagerServiceImpl,ClientCryptoManagerService clientCryptoManagerService,Context context) {
+        this.registrationService = registrationService;
+        this.context = context;
+        this.fileSignatureRepository = fileSignatureRepository;
+        this.applicationMap = new HashMap<>();
+        this.globalParamRepository = globalParamRepository;
+        this.masterDataService = masterDataService;
+        this.certificateManagerService = certificateManagerService;
+        this.cryptoManagerServiceImpl = cryptoManagerServiceImpl;
+        this.clientCryptoManagerService = clientCryptoManagerService;
+    }
+    @Override
+    public void getDocumentCategories(@NonNull String categoryCode, @NonNull String langCode, @NonNull DocumentCategoryPigeon.Result<List<String>> result) {
+        List<String> documentCategory = new ArrayList<>();
+        try {
+            Map<String, Object> dataContext = this.registrationService.getRegistrationDto().getMVELDataContext();
+            String applicantTypeCode = this.evaluateMvelScript((String) this.globalParamRepository.getCachedStringMAVELScript(), dataContext);
+            documentCategory = this.masterDataService.getDocumentTypes(categoryCode, applicantTypeCode, langCode);
+        } catch (Exception e) {
+            Log.e(getClass().getSimpleName(), "Fetch document values: " + Arrays.toString(e.getStackTrace()));
+        }
+        result.success(documentCategory);
+    }
+
+
+    public String evaluateMvelScript(String scriptName, Map<String, Object> dataContext) {
+        try {
+            Map<String, String>  ageGroups = new HashMap<String, String>();
+            JSONObject ageGroupConfig = new JSONObject((String) this.globalParamRepository.getCachedStringAgeGroup());
+            for (Iterator<String> it = ageGroupConfig.keys(); it.hasNext(); ) {
+                String key = it.next();
+                ageGroups.put(key, ageGroupConfig.getString(key));
+            }
+
+            Map context = new HashMap();
+            MVEL.eval(getScript(scriptName), context);
+            context.put("identity", dataContext);
+            context.put("ageGroups", ageGroups);
+            return MVEL.eval("return getApplicantType();", context, String.class);
+        } catch (Throwable t) {
+            Log.e(getClass().getSimpleName(),"Failed to evaluate mvel script", t);
+        }
+        return null;
```

Comment:
returning null and having no check where it's called is error prone

Reply:
we are already handling null exception in the usage methods @jainhitesh9998